### ISSUE TITLE
Automatic reopening of opened gumps on login.

### ISSUE THREAD

**Issue by ZaneDubya** [ZaneDubya/UltimaXNA#163]
@Eskymak said, "Could we also get automatic reopening of last (before logout) opened gumps, after login?"

I imagine this would be comprised of two parts: first, a Settings that saved a list of open gumps (perhaps by type? or by name?); second, a flag that would only be true on local gumps so the client doesn't attempt to open server-created gumps (like bank boxes, corpse contents, npc paperdolls, etc.) on login.

**Comment by fdsprod** [comment]
Gumps come in 2 types, server (contains a serial) and client (no serial).  Most client gumps are also singletons.  So it would be best to save off a list of gump types that are open that do not contain a Serial.

**Comment by ZaneDubya** [comment]
Implementation dependent on #210.

**Comment by ZaneDubya** [comment]
Ok, let's get this done!

**Comment by ZaneDubya** [comment]
Mostly done in 35eecf504c4b2afa1b59854f9cf4c99ce605575f.
Note that enabling this behavior for gumps will require rewriting the gump's ctor and overriding the new SaveGump/LoadGump methods.

**Comment by ZaneDubya** [comment]
Complete. Note that I've only implemented the code for the PaperDollGump, you will need to make the requisite changes in each other type of gump to allow this new code to restore them as well. If you would like me to do it, open a new issue for that gump type. ;)
